ngraph/frontend/onnx_import/src/op/constant_fill.cpp

```diff
@@ -0,0 +1,67 @@
+// Copyright (C) 2018-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <onnx/onnx_pb.h> // onnx types
+
+#include "default_opset.hpp"
+#include "exceptions.hpp"
+#include "ngraph/op/broadcast.hpp"
+#include "ngraph/op/concat.hpp"
+#include "ngraph/op/constant.hpp"
+#include "onnx_common/utils.hpp"
+
+namespace ngraph
+{
+    namespace onnx_import
+    {
+        namespace op
+        {
+            namespace set_1
+            {
+                // ConstantFill is a deprecated experimental operator removed in ONNX 1.4
+                OutputVector constant_fill(const Node& node)
+                {
+                    Output<ngraph::Node> target_shape;
+                    const auto fill_value = node.get_attribute_value<float>("value", 0.f);
+                    const auto dtype = node.get_attribute_value<int64_t>(
+                        "dtype", static_cast<int64_t>(ONNX_NAMESPACE::TensorProto_DataType_FLOAT));
+                    const auto ng_type = onnx_common::onnx_to_ng_data_type(
+                        static_cast<ONNX_NAMESPACE::TensorProto_DataType>(dtype));
+                    const auto const_val_to_fill = std::make_shared<default_opset::Convert>(
+                        default_opset::Constant::create(element::f32, {}, {fill_value}), ng_type);
+                    const auto input_as_shape =
+                        node.get_attribute_value<int64_t>("input_as_shape", 1);
+                    if (input_as_shape == 1) // use the first input as target shape
+                    {
+                        CHECK_VALID_NODE(
+                            node,
+                            node.get_ng_inputs().size() > 0,
+                            "The input which determines output shape was not provided");
+                        target_shape = node.get_ng_inputs().at(0);
+                        if (node.has_attribute("extra_shape"))
+                        {
+                            const auto extra_shape =
+                                node.get_attribute_value<std::vector<int64_t>>("extra_shape");
+                            const auto extra_shape_const = default_opset::Constant::create(
+                                target_shape.get_element_type(), {extra_shape.size()}, extra_shape);
```

Comment:
I wonder if function such as `get_attribute_as_constant` is a good idea.
What do you think?

Reply:
Discussed offline. Created a separated ticket: 53284